profile/main.go

```diff
@@ -282,16 +282,19 @@ func getdata(client *firestore.Client, ctx context.Context, userId string, userU
 	var res Res
 	json.Unmarshal([]byte(r), &res)
 
-	lastdiff, lastdiffId := getLastDiff(client, ctx, userId)
+	lastPendingDiff, lastPendingDiffId := getLastDiff(client, ctx, userId, "PENDING")
 	userData := getUserData(client, ctx, userId)
-	if lastdiff != res && userData != res {
-		if lastdiffId != "" {
-			setNotApproved(client, ctx, lastdiffId)
+	if lastPendingDiff != res && userData != res {
+		if lastPendingDiffId != "" {
+			setNotApproved(client, ctx, lastPendingDiffId)
+		}
+		lastRejectedDiff, _ := getLastDiff(client, ctx, userId, "NOT APPROVED")
+		if lastRejectedDiff != res {
+			generateAndStoreDiff(client, ctx, res, userId)
 		}
```

Comment:
let's log it in the else part with
type: "SAME_PROFILE_DIFF"
timestamp: <timestamp>
meta: { profileDiffId: <lastDiffRejected> }
body: { profileDiffId: <lastDiffRejected> }

Reply:
Changes are done